sqlalchemy_exasol/base.py

```diff
@@ -661,63 +667,85 @@ def get_view_definition_sql(self, connection, view_name, schema=None, **kw):
         else:
             return None
 
+    def get_column_sql_query_str(self):
+        return "SELECT " \
+               "column_name, " \
+               "column_type, " \
+               "column_maxsize, " \
+               "column_num_prec, " \
+               "column_num_scale, " \
+               "column_is_nullable, " \
+               "column_default, " \
+               "column_identity, " \
+               "column_is_distribution_key, " \
+               "column_table " \
+               "FROM sys.exa_all_columns " \
+               "WHERE " \
+               "column_object_type IN ('TABLE', 'VIEW') AND " \
+               "column_schema = {schema} AND " \
+               "column_table = {table} " \
+               "ORDER BY column_ordinal_position"
+
     @reflection.cache
-    def _get_all_columns_odbc(self, connection, odbc_connection, schema, **kw):
+    def _get_all_columns_odbc(self, connection, odbc_connection, table_name, schema, **kw):
         schema = self._get_schema_for_input_or_current(connection, schema)
         # We need to check if the schema is valid
-        tables = self._get_tables_for_schema_odbc(connection, odbc_connection, schema=schema)
+        tables = self._get_tables_for_schema_odbc(connection, odbc_connection,
+                                                  schema=schema, table_name=table_name, **kw)
 
-        if len(tables) > 0:
+        if len(tables) == 1:
```

Comment:
Why was this if condition changed? What happens if len(tables) = 2 or greater?

Reply:
len(tables) = 2 shouldn't be possible anymore. However, we should at least check it and raise an exception if it happens. Furthermore, I see we need to rename the methods to `get_columns_of_table`, because the SQL we use to fetch the columns now also filters by table_name.